dataprofiler/tests/profilers/profiler_options/test_unstructured_text_profile_options.py

```diff
@@ -244,3 +247,57 @@ def test_eq(self):
         self.assertNotEqual(options, options2)
         options2.words.is_enabled = False
         self.assertEqual(options, options2)
+
+    def test_json_encode(self):
+        option = TextProfilerOptions(
+            is_enabled=False,
+            is_case_sensitive=False,
+            stop_words={"ab", "aa", "aba", "aa"},
+            top_k_chars=5,
+            top_k_words=8,
+        )
+        serialized = json.dumps(option, cls=ProfileEncoder)
+
+        expected_class = "TextProfilerOptions"
+        expected_options_attributes = {
+            "is_enabled",
+            "is_case_sensitive",
+            "stop_words",
+            "top_k_chars",
+            "top_k_words",
+            "vocab",
+            "words",
+        }
+        expected_is_enabled = option.is_enabled
+        expected_is_case_sensitive = option.is_case_sensitive
+        expected_stop_words = option.stop_words
+        expected_top_k_chars = option.top_k_chars
+        expected_top_k_words = option.top_k_words
+
+        actual_option_json = json.loads(serialized)
+
+        self.assertIn("class", actual_option_json)
```

Comment:
I think this got significantly more complicated than necessary, but does achieve the tests. maybe move forward and refactor in a subsequent PR or remove with deserial since this feels duplicative in nature.